onadata/libs/utils/analytics.py

```diff
@@ -143,12 +154,36 @@ def get_event_label(self) -> str:
             event_label = f"form-{form_id}-owned-by-{username}"
         return event_label
 
+    def get_request_origin(self, request, tracking_properties):
+        if isinstance(self.tracked_obj, Instance):
+            try:
+                user_agent = request.META['HTTP_USER_AGENT']
+                if 'Android' in user_agent:
+                    event_source = f'Submission collected from ODK COLLECT'
+                elif 'Chrome' or 'Mozilla' or 'Safari' in user_agent:
+                    event_source = f'Submission collected from Enketo'
+            except KeyError:
+                event_source = ""
+        else:
+            event_source = ""
+        tracking_properties.update({'event_source': event_source})
+        return tracking_properties
+
     def _track_object_event(self, obj, request=None) -> None:
         self.tracked_obj = obj
         self.set_user()
         event_name = self.get_event_name()
         label = self.get_event_label()
         tracking_properties = self.get_tracking_properties(label=label)
+        try:
+            if tracking_properties['from'] == 'XML Submissions':
+                # Only introduce an `event_source` field for submissions
+                # created xml. This helps differentiate Enketo and ODK Collect
+                # instances. Otherwise, use the `from` tag in properties object
+                tracking_properties = self.get_request_origin(
```

Comment:
I still don't see the need to have both the `from` and `event_source` fields... Both refer to the same aspect of an event. I'd suggest we use the `from` field as that's already configured in Segment

Reply:
Changed this, remained with the `event_source` tag.